netbox_onboarding/onboarding/onboarding.py

```diff
@@ -0,0 +1,34 @@
+"""Onboarding module.
+
+(c) 2020 Network To Code
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+  http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from netbox_onboarding.netbox_keeper import NetboxKeeper
+
+
+class Onboarding(object):
+    """Generic onboarding class."""
+
+    def __init__(self):
+        """Init the class."""
+        self.created_device = None
+
```

Comment:
I think it would makes sense to create a default implementation of `run` in this base class, which would be simply:

```python
def run(self, onboarding_kwargs):
    nb = NetboxKeeper(**onboarding_kwargs)
    nb.ensure_device()
    self.created_device = nb.device
```

That way both `StandaloneOnboarding` and the example `MyOnboardingClass` could have a `run` implementation that looks like:

```python
def run(self, onboarding_kwargs):
    # do any special work here to populate or alter onboarding_kwargs, then:
    super().run(onboarding_kwargs)
```

and classes that need to do something special (like #67) would simply not include the `super().run()` call.

Reply:
I agree with your intention - every sub-class implementation I saw had to customize the run() method, so I would rather have it like this:
```
class Onboarding:
    """Generic onboarding class."""

    def __init__(self):
        """Init the class."""
        self.created_device = None

    def run(self):
        raise NotImplementedError
```